Modify respond_to behaviour always setting the response's content type based on the request format (#37617)
rails/rails: guides/source/upgrading_ruby_on_rails.md

To allow you to upgrade to new defaults one by one, the update task has created a file `config/initializers/new_framework_defaults.rb`. Once your application is ready to run with new defaults, you can remove this file and flip the `config.load_defaults` value.

Upgrading from Rails 6.0 to Rails 6.1
-------------------------------------

### Response's Content-Type when using `respond_to#any`

The Content-Type header returned in the response can differ from what Rails 6.0 returned,
more specifically if your application uses `respond_to { |format| format.any }`.
The Content-Type will now be based on the given block rather than the request's format.

Example:

```ruby
  def my_action
    respond_to do |format|
      format.any { render(json: { foo: 'bar' }) }
    end
  end

  get('my_action.csv')
```

> Previous behaviour was returning a `text/csv` response's Content-Type which is innacurate since w are returning a JSON response.

Reviewer: ```suggestion
Previous behaviour was returning a `text/csv` response's Content-Type which is inaccurate since a JSON response is being rendered.
```
